Scripts to Test MC 22-23 Production (UL-like) in IBs

AdrianoDee opened the pull request:
This PR proposes the addition of a set of scripts to be used as IB tests to show the desired behavior of the next 2022-2023 MC campaign and to allow experts to have this run in all the IBs to follow issues and eventually find solutions. The schema for the campaign should be the following:

#### 2022 campaign
- GEN-SIM-RAW-DIGI steps run in `14_0_X`;
- HLT step for 2022 run in `12_4_X`;
- all the subsequent steps run in `14_0_X`;
- ideal pre EE issue conditions. The EE issue to be mimicked via a veto.

#### 2023 campaign
- GEN-SIM-RAW-DIGI steps run in `14_0_X`;
- HLT step for 2023 run in `13_0_X`;
- all the subsequent steps run in `14_0_X`;
- ideal pre BPix hole conditions. The BPix issue to be absorbed in the SF.

This setup poses some issues:
- forward compatibility: CMSSW does not formally support it so `12_4_X` or `13_0_X` being able to process `14_0_X` samples is not granted. 
- conditions: we need to be able to have exactly the same conditions for [`140X_*preEE*` and `124X_*preEE*`] and [`140X_*preBPix*` and `130X_*preBPix*`]. Possibly having the `SimBeamSpotObjectsRcd` in them, to be able to use the BS smearing from GT mechanism at GEN-SIM step.

To make it less cumbersome (hopefully), I've split each step in a different script and have the different setups running the same bash script with different inputs. **As they are, none of the tests technically fails**. This is intentional not to have some tests that will by definition break any IB. But any suggestion there is welcome. Each script runs the HLT step in a specific target release and all the steps in the current release. 

_Issue (?): as they are these tests take O(100s) each (so x4)._ 

We would need a back port to `14_0_X` to have it tested there.

FYI: @cms-sw/alca-l2, @cms-sw/core-l2

Commit:
Script to test MC production (UL-like)
- added specific tests for 2022-2023
- added a general script to test a setup where HLT is not run in the current release

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"3e1901da5c0d70b9f88db251feb8c9987ccac878":{"files":["Configuration/PyReleaseValidation/scripts/test-MC-setup_gen_sim.sh","Configuration/PyReleaseValidation/scripts/test-MC-setup_hlt.sh","Configuration/PyReleaseValidation/scripts/test-MC-setup_reco.sh","Configuration/PyReleaseValidation/test/BuildFile.xml","Configuration/PyReleaseValidation/test/test_mc_setup/test-MC_22_setup.sh","Configuration/PyReleaseValidation/test/test_mc_setup/test-MC_22_setup_crosscheck.sh","Configuration/PyReleaseValidation/test/test_mc_setup/test-MC_23_setup.sh","Configuration/PyReleaseValidation/test/test_mc_setup/test-MC_23_setup_crosscheck.sh","Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh"],"squashed":false,"time":1711719186}},"emoji":{},"last_seen_sha":"3e1901da5c0d70b9f88db251feb8c9987ccac878","signatures":{}} -->

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/39736

- This PR adds an extra 20KB to repository

cmsbuild commented:
New categories assigned: core

@Dr15Jones,@makortel,@smuzaffar you have been requested to review this Pull request/Issue and eventually sign? Thanks

cmsbuild commented:
A new Pull Request was created by  @AdrianoDee for master.

It involves the following packages:

- Configuration/PyReleaseValidation (**upgrade, pdmv**)


@AdrianoDee, @Dr15Jones, @smuzaffar, @sunilUIET, @makortel, @miquork, @cmsbuild, @subirsarkar, @srimanob can you please review it and eventually sign? Thanks.
@Martin-Grunewald, @missirol, @makortel, @fabiocos, @slomeo this is something you requested to watch as well.
@antoniovilela, @rappoccio, @sextonkennedy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

AdrianoDee commented:
please test

mmusich commented on Configuration/PyReleaseValidation/scripts/test-MC-setup_gen_sim.sh:
```diff
@@ -0,0 +1,24 @@
+era=$3
+beamspot=$4
+
+echo '> Running GEN,SIM + DIGI,L1,DIGI2RAW stes in ' $release
```
```suggestion
echo '> Running GEN,SIM + DIGI,L1,DIGI2RAW steps in ' $release
```

mmusich commented on Configuration/PyReleaseValidation/scripts/test-MC-setup_gen_sim.sh:
```diff
@@ -0,0 +1,24 @@
+--fileout file:step2.root --conditions $conditions --step DIGI,L1,DIGI2RAW \
+--geometry DB:Extended --era $era --mc -n -1
+
+############################################################################################################
```
missing newline

mmusich commented on Configuration/PyReleaseValidation/scripts/test-MC-setup_reco.sh:
```diff
@@ -0,0 +1,27 @@
+--conditions $conditions --step PAT \
+--geometry DB:Extended --filein file:step4.root --era $era \
+--runUnscheduled --mc -n -1
+############################################################################################################
```
missing newline

mmusich commented on Configuration/PyReleaseValidation/test/test_mc_setup/test-MC_22_setup.sh:
```diff
@@ -0,0 +1,12 @@
+
+$CMSSW_BASE/src/Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh \
+140X_mcRun3_2022_realistic_postEE_v3 Run3 2022v14 CMSSW_12_4_19 \
+124X_mcRun3_2022_realistic_postEE_v1 Realistic25ns13p6TeVEarly2022Collision
```
missing newline

mmusich reviewed (commented):
a couple of typos (also missing newlines since there was recently a campaign to get rid of them).

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/39738

- This PR adds an extra 16KB to repository

AdrianoDee commented on Configuration/PyReleaseValidation/test/test_mc_setup/test-MC_23_setup_crosscheck.sh:
```diff
@@ -0,0 +1,12 @@
+
+$CMSSW_BASE/src/Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh \
+140X_mcRun3_2023_realistic_postBPix_v3 Run3_2023 @relval2023 $CMSSW_VERSION \
+140X_mcRun3_2023_realistic_postBPix_v3 DBrealistic
```
I should have added them now (I see them locally) but they don't show up here (and I don't see anymore the warning from GitHub). Is this is a GitHub feature?

cmsbuild commented:
Pull request #44578 was updated. @Dr15Jones, @AdrianoDee, @makortel, @smuzaffar, @cmsbuild, @srimanob, @subirsarkar, @sunilUIET, @miquork can you please check and sign again.

AdrianoDee commented:
please test

mmusich commented on Configuration/PyReleaseValidation/scripts/test-MC-setup_hlt.sh:
```diff
@@ -0,0 +1,33 @@
+
+function rel_setup()
+{
+    scram p CMSSW $1
```
out of curiosity, since this is "fixed" instead of creating a new checkout area would it not be possible to just do `cmsenv` inside the area `/cvmfs/cms.cern.ch/$hlt_arch/cms/cmssw/$1/` ?

AdrianoDee replied:
I think so, actually seems cleaner too. Let me try.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10f471/38511/summary.html
**COMMIT**: ac1db4c8a946abd7fc4118c35bbd64c9525c0761
**CMSSW**: CMSSW_14_1_X_2024-03-28-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/44578/38511/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 108 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-03-28-2300+10f471/62031/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 51 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-03-28-2300+10f471/62031/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3297469
 - DQMHistoTests: [Total failures: 6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-03-28-2300+10f471/62031/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3297443
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-03-28-2300+10f471/62031/triggerResults): no differences found

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/39749

- This PR adds an extra 16KB to repository

cmsbuild commented:
Pull request #44578 was updated. @subirsarkar, @makortel, @AdrianoDee, @smuzaffar, @cmsbuild, @Dr15Jones, @miquork, @sunilUIET, @srimanob can you please check and sign again.

AdrianoDee commented:
please test

makortel commented on Configuration/PyReleaseValidation/scripts/test-MC-setup_gen_sim.sh:
How necessary it is for these scripts to be in `$PATH`?

AdrianoDee replied:
Actually it's only to avoid copying them in the curent test folder. Don't know which solution is cleaner.

makortel replied:
Ok. I think the `test/` would be a better place then. You can use `$SCRAM_TEST_PATH/<script name>.sh` to access the scripts from the other scripts.

makortel commented on Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh:
```diff
@@ -0,0 +1,105 @@
+echo ''
+
+# the tmp dir cleaning 
+function job_clean()
```
Why is the cleanup useful? We don't generally clean up test artifacts, and in some cases having them around after a (failed) script is useful for debugging.

AdrianoDee replied:
All the cleanup was connected with https://github.com/cms-sw/cmssw/pull/44578#discussion_r1547337493, the idea I needed to be outside the "default" testing folder and then the fact I thought it was needed to clean up this extra repository.

makortel commented on Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh:
```diff
@@ -0,0 +1,105 @@
+############################################################################################################
+# RECO + PAT
+
+cd $CMSSW_BASE
```
Doesn't `$CMSSW_BASE` point to `/cvmfs/cms.cern.ch/$hlt_arch/cms/cmssw/$release_hlt/` at this point?

AdrianoDee replied:
Yes, fixing it.

makortel commented on Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh:
```diff
@@ -0,0 +1,105 @@
+
+############################################################################################################
+# setting up the temporary working dir, to be removed once done (or crashed)
+tmp_dir=$(mktemp -d -t test_MC_${3}_${4}_${5}_XXXXXXX)
```
Scram already runs each unit test in its own unique directory, so setting up a(nother) temporary directory is not strictly needed?

AdrianoDee replied:
Indeed not really. I thought, since I want to checkout a new release, I needed to be outside the current `$CMSSW_BASE`. But evidently I was mistaken.

makortel commented on Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh:
```diff
@@ -0,0 +1,105 @@
+# to take into account changes in gcc
+base_arch_no_gcc=$(echo $SCRAM_ARCH | cut -d "_" -f -2)
+if [[ ! ("$CMSSW_VERSION" == $release_hlt) ]]; then
+    hlt_arch=$(echo $(ls -dtrh /cvmfs/cms.cern.ch/${base_arch_no_gcc}*/cms/cmssw/${release_hlt} | tail -1) | cut -d "/" -f 4)
```
@smuzaffar Is there a better way to figure out the installation area of "the other release" than hardcoding a dependence on CVMFS?

davidlange6 replied:
perhaps using $CMS_PATH [indeed, building in cvmfs assumptions is best avoided]

smuzaffar replied:
one can use scram itself to find the cmssw release e.g.
```
scram list -c $CMSSW_VERSION | grep $CMSSW_VERSION | sed 's|.* ||'
```

note that this test should be run conditionally as not all hlt releases will be available for all `base_arch_no_gcc`

AdrianoDee replied:
> `scram list -c $CMSSW_VERSION | grep $CMSSW_VERSION | sed 's|.* ||'`

Isn't this assuming I know the full `SCRAM_ARCH` for the target HLT release? Here I was trying to infer the newest available gcc for the target release. I agree I would need to add a check that such a release is actually available.

makortel reviewed (commented):
First pass

makortel commented on Configuration/PyReleaseValidation/test/test_mc_setup/test-MC_23_setup.sh:
```diff
@@ -0,0 +1,12 @@
+echo ""
+
+$CMSSW_BASE/src/Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh \
+140X_mcRun3_2023_realistic_v3 Run3_2023 2023v12 CMSSW_13_0_17 \
```
Can this use 13_0_18? That includes https://github.com/cms-sw/cmsdist/pull/9017 that should get the job beyond
```
----- Begin Fatal Exception 29-Mar-2024 17:11:50 CET-----------------------
An exception of category 'FileOpenError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing input source of type PoolSource
   [2] Calling RootInputFileSequence::initTheFile()
   Additional Info:
      [a] Input file file:step2.root could not be opened.
      [b] Fatal Root Error: @SUB=TList::Clear
A list is accessing an object (0x14b339b8b740) already deleted (list name = TList)

----- End Fatal Exception -------------------------------------------------
```
and demonstrate the next failure (that we are going to address)

AdrianoDee replied:
Sure, let me fix it.

makortel commented:
> **As they are, none of the tests technically fails**. This is intentional not to have some tests that will by definition break any IB. But any suggestion there is welcome.

I'd prefer the scripts would be crafted such that they fail in case the any of the jobs fail, so that the failures are clearly demonstrated in the PR tests.

In a way I'd also prefer to integrate the failed tests to the IBs, but I acknowledge having two tests failing there for a week or two is not ideal and could hide other problems that arise in the mean time. Maybe some simple switch to make the script ignore the errors of the HLT step, that could be removed after things work.

There are couple of additional details (beyond this PR) that should be discussed in (e.g.) ORP at some point.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10f471/38528/summary.html
**COMMIT**: 61c70e5de133b1ccfbfab99ebc9adf592f748105
**CMSSW**: CMSSW_14_1_X_2024-04-01-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/44578/38528/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 3 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-01-1100+10f471/62043/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 45 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-01-1100+10f471/62043/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3297469
 - DQMHistoTests: [Total failures: 3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-01-1100+10f471/62043/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3297446
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-01-1100+10f471/62043/triggerResults): no differences found

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/39763

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - Configuration/PyReleaseValidation/test/test_mc_setup/test-MC_22_setup_crosscheck.sh:
    - Added: ac1db4c8a946abd7fc4118c35bbd64c9525c0761
    - Deleted: d3cb952c414c1eec4ad24b2d911eaf5ec804ed3b
  - Configuration/PyReleaseValidation/test/test_mc_setup/test-MC_23_setup.sh:
    - Added: ac1db4c8a946abd7fc4118c35bbd64c9525c0761
    - Deleted: d3cb952c414c1eec4ad24b2d911eaf5ec804ed3b
  - Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh:
    - Added: ac1db4c8a946abd7fc4118c35bbd64c9525c0761
    - Modified: 61c70e5de133b1ccfbfab99ebc9adf592f748105
    - Deleted: d3cb952c414c1eec4ad24b2d911eaf5ec804ed3b
  - Configuration/PyReleaseValidation/test/test_mc_setup/test-MC_23_setup_crosscheck.sh:
    - Added: ac1db4c8a946abd7fc4118c35bbd64c9525c0761
    - Deleted: d3cb952c414c1eec4ad24b2d911eaf5ec804ed3b
  - Configuration/PyReleaseValidation/scripts/test-MC-setup_reco.sh:
    - Added: ac1db4c8a946abd7fc4118c35bbd64c9525c0761
    - Modified: 61c70e5de133b1ccfbfab99ebc9adf592f748105
    - Deleted: d3cb952c414c1eec4ad24b2d911eaf5ec804ed3b
  - Configuration/PyReleaseValidation/test/test_mc_setup/test-MC_22_setup.sh:
    - Added: ac1db4c8a946abd7fc4118c35bbd64c9525c0761
    - Deleted: d3cb952c414c1eec4ad24b2d911eaf5ec804ed3b
  - Configuration/PyReleaseValidation/scripts/test-MC-setup_gen_sim.sh:
    - Added: ac1db4c8a946abd7fc4118c35bbd64c9525c0761
    - Deleted: d3cb952c414c1eec4ad24b2d911eaf5ec804ed3b
  - Configuration/PyReleaseValidation/scripts/test-MC-setup_hlt.sh:
    - Added: ac1db4c8a946abd7fc4118c35bbd64c9525c0761
    - Modified: 61c70e5de133b1ccfbfab99ebc9adf592f748105
    - Deleted: d3cb952c414c1eec4ad24b2d911eaf5ec804ed3b

cmsbuild commented:
Pull request #44578 was updated. @srimanob, @miquork, @subirsarkar, @Dr15Jones, @makortel, @sunilUIET, @smuzaffar, @AdrianoDee, @cmsbuild can you please check and sign again.

AdrianoDee commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/39764

- This PR adds an extra 12KB to repository

cmsbuild commented:
Pull request #44578 was updated. @Dr15Jones, @smuzaffar, @AdrianoDee, @sunilUIET, @srimanob, @makortel, @subirsarkar, @miquork can you please check and sign again.

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10f471/38552/summary.html
**COMMIT**: d4a9629481617a71902c4de0619f4433646f0e4c
**CMSSW**: CMSSW_14_1_X_2024-04-02-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/44578/38552/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 4 errors in the following unit tests:

<pre>---> test test-MC_22_crosscheck had ERRORS
---> test test-MC_23_crosscheck had ERRORS
---> test test-MC_23_setup had ERRORS
and more ...
</pre>

## Comparison Summary



**Summary**:
 - You potentially [removed 78 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-02-1100+10f471/62064/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 43 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-02-1100+10f471/62064/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3297469
 - DQMHistoTests: [Total failures: 0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-02-1100+10f471/62064/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3297449
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-02-1100+10f471/62064/triggerResults): no differences found

AdrianoDee commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/39781

- This PR adds an extra 16KB to repository

cmsbuild commented:
Pull request #44578 was updated. @sunilUIET, @smuzaffar, @makortel, @miquork, @srimanob, @AdrianoDee, @subirsarkar, @Dr15Jones can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/39790

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - Configuration/PyReleaseValidation/test/test-MC-setup_hlt.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
  - Configuration/PyReleaseValidation/test/test_MC_setup.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
  - Configuration/PyReleaseValidation/test/test-MC-setup_reco.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
  - Configuration/PyReleaseValidation/test/test-MC-setup_gen_sim.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370

cmsbuild commented:
Pull request #44578 was updated. @miquork, @subirsarkar, @sunilUIET, @makortel, @smuzaffar, @AdrianoDee, @cmsbuild, @srimanob, @Dr15Jones can you please check and sign again.

AdrianoDee commented:
please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10f471/38579/summary.html
**COMMIT**: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
**CMSSW**: CMSSW_14_1_X_2024-04-03-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/44578/38579/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test test_MC_23_setup had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially [removed 108 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-03-1100+10f471/62091/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 42 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-03-1100+10f471/62091/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3304819
 - DQMHistoTests: [Total failures: 0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-03-1100+10f471/62091/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3304799
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-03-1100+10f471/62091/triggerResults): no differences found

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/39794

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - Configuration/PyReleaseValidation/test/test-MC-setup_hlt.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
  - Configuration/PyReleaseValidation/test/test_MC_setup.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
  - Configuration/PyReleaseValidation/test/test-MC-setup_reco.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
  - Configuration/PyReleaseValidation/test/test-MC-setup_gen_sim.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370

cmsbuild commented:
Pull request #44578 was updated. @miquork, @srimanob, @makortel, @Dr15Jones, @sunilUIET, @smuzaffar, @AdrianoDee, @subirsarkar can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/39796

- This PR adds an extra 20KB to repository

- Found files with invalid states:
  - Configuration/PyReleaseValidation/test/test-MC-setup_hlt.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
  - Configuration/PyReleaseValidation/test/test_MC_setup.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
  - Configuration/PyReleaseValidation/test/test-MC-setup_reco.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
  - Configuration/PyReleaseValidation/test/test-MC-setup_gen_sim.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370

cmsbuild commented:
Pull request #44578 was updated. @cmsbuild, @miquork, @Dr15Jones, @sunilUIET, @smuzaffar, @AdrianoDee, @makortel, @srimanob, @subirsarkar can you please check and sign again.

AdrianoDee commented:
please test

AdrianoDee commented:
please abort

AdrianoDee commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/39801

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - Configuration/PyReleaseValidation/test/test-MC-setup_hlt.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
  - Configuration/PyReleaseValidation/test/test_MC_setup.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
  - Configuration/PyReleaseValidation/test/test-MC-setup_reco.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370
  - Configuration/PyReleaseValidation/test/test-MC-setup_gen_sim.sh:
    - Added: 48ea677ba9a83005b33c9161ebc25551d0933f82
    - Deleted: 7effb6fb3cf31b680b3a6b5273eaf8019de6b370

cmsbuild commented:
Pull request #44578 was updated. @Dr15Jones, @smuzaffar, @subirsarkar, @makortel, @AdrianoDee, @miquork, @srimanob, @sunilUIET can you please check and sign again.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10f471/38597/summary.html
**COMMIT**: 7544e0f050068c9bc0a683e31c992568282c2d1b
**CMSSW**: CMSSW_14_1_X_2024-04-03-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/44578/38597/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



There are some workflows for which there are errors in the baseline:
 24834.78 step 2
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

**Summary**:
 - You potentially [added 75 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-03-2300+10f471/62098/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 42 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-03-2300+10f471/62098/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3304819
 - DQMHistoTests: [Total failures: 3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-03-2300+10f471/62098/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3304796
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-03-2300+10f471/62098/triggerResults): no differences found

AdrianoDee commented:
A (late) summary of the tests here:

- following @davidlange6 suggestion, I've moved the dependency on `cvmfs` to `CMSSW_PATH` to search for the latest gcc available for the HLT CMSSW;
- the tests as they are don't trigger a failure at HLT step unless the `CMSSW_MC_SETUP_TEST_CATCH_HLT` environment variable is set;
- the [tests here](https://github.com/cms-sw/cmssw/pull/44578#issuecomment-2035388452) shows the behavior when the variable is set for the 2023 MC setup)

I would wait the end of the review to squash the commits.

mmusich commented:
> /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_14_1_X_2024-04-03-2300/src/Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh: line 66: cd: /cvmfs/cms-ib.cern.ch//cms/cmssw/CMSSW_12_4_19/: No such file or directory

in [log](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10f471/38597/unitTests/src/Configuration/PyReleaseValidation/test/test_MC_22_setup/testing.log). Is this the desired behaviour?

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/39858

- This PR adds an extra 12KB to repository

- Found files with invalid states:
  - Configuration/PyReleaseValidation/test/test-MC-setup_hlt.sh:
    - Added: 92cda626a7c55d6aaf6877c74ef38376115cf07c
    - Deleted: 1a48a39d49c28059f0b068d1972aa37b960075ef
  - Configuration/PyReleaseValidation/test/test_MC_setup.sh:
    - Added: 92cda626a7c55d6aaf6877c74ef38376115cf07c
    - Deleted: 1a48a39d49c28059f0b068d1972aa37b960075ef
  - Configuration/PyReleaseValidation/test/test-MC-setup_reco.sh:
    - Added: 92cda626a7c55d6aaf6877c74ef38376115cf07c
    - Deleted: 1a48a39d49c28059f0b068d1972aa37b960075ef
  - Configuration/PyReleaseValidation/test/test-MC-setup_gen_sim.sh:
    - Added: 92cda626a7c55d6aaf6877c74ef38376115cf07c
    - Deleted: 1a48a39d49c28059f0b068d1972aa37b960075ef

cmsbuild commented:
Pull request #44578 was updated. @sunilUIET, @AdrianoDee, @makortel, @smuzaffar, @miquork, @Dr15Jones, @subirsarkar, @cmsbuild, @srimanob can you please check and sign again.

AdrianoDee commented:
> > /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_14_1_X_2024-04-03-2300/src/Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh: line 66: cd: /cvmfs/cms-ib.cern.ch//cms/cmssw/CMSSW_12_4_19/: No such file or directory
> 
> in [log](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10f471/38597/unitTests/src/Configuration/PyReleaseValidation/test/test_MC_22_setup/testing.log). Is this the desired behaviour?

Err, no. And this is the proof that I need a better method to infer the right architecture.

cmsbuild commented:
Pull request #44578 was updated. @srimanob, @Dr15Jones, @AdrianoDee, @miquork, @subirsarkar, @cmsbuild, @makortel, @sunilUIET, @smuzaffar can you please check and sign again.

AdrianoDee commented:
please test
Next test should fail by construction.

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10f471/38687/summary.html
**COMMIT**: fb4419879c023e37639c55d0905a32cd868a79ac
**CMSSW**: CMSSW_14_1_X_2024-04-08-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/44578/38687/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 2 errors in the following unit tests:

<pre>---> test test_MC_22_setup had ERRORS
---> test test_MC_23_setup had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially [removed 110 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-08-1100+10f471/62185/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 44 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-08-1100+10f471/62185/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3307717
 - DQMHistoTests: [Total failures: 6](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-08-1100+10f471/62185/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3307691
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-08-1100+10f471/62185/triggerResults): no differences found

AdrianoDee commented:
please test
- these tests should succeed instead.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/39861

- This PR adds an extra 16KB to repository

- Found files with invalid states:
  - Configuration/PyReleaseValidation/test/test-MC-setup_hlt.sh:
    - Added: 92cda626a7c55d6aaf6877c74ef38376115cf07c
    - Deleted: fb4419879c023e37639c55d0905a32cd868a79ac
  - Configuration/PyReleaseValidation/test/test_MC_setup.sh:
    - Added: 92cda626a7c55d6aaf6877c74ef38376115cf07c
    - Deleted: fb4419879c023e37639c55d0905a32cd868a79ac
  - Configuration/PyReleaseValidation/test/test-MC-setup_reco.sh:
    - Added: 92cda626a7c55d6aaf6877c74ef38376115cf07c
    - Deleted: fb4419879c023e37639c55d0905a32cd868a79ac
  - Configuration/PyReleaseValidation/test/test-MC-setup_gen_sim.sh:
    - Added: 92cda626a7c55d6aaf6877c74ef38376115cf07c
    - Deleted: fb4419879c023e37639c55d0905a32cd868a79ac

cmsbuild commented:
Pull request #44578 was updated. @Dr15Jones, @subirsarkar, @miquork, @smuzaffar, @srimanob, @makortel, @sunilUIET, @AdrianoDee can you please check and sign again.

makortel commented:
> ---> test test_MC_22_setup had ERRORS
> ---> test test_MC_23_setup had ERRORS

Thanks, these failures look as expected for these tests.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10f471/38689/summary.html
**COMMIT**: 0c89fb1ef5aabc242ec03996fc7dc5bdba23d119
**CMSSW**: CMSSW_14_1_X_2024-04-08-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/44578/38689/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 87 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-08-1100+10f471/62187/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 45 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-08-1100+10f471/62187/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3307717
 - DQMHistoTests: [Total failures: 3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-08-1100+10f471/62187/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3307694
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-08-1100+10f471/62187/triggerResults): no differences found

AdrianoDee commented:
So a recap, this time with the right results:
- following @smuzaffar  suggestion (sorry for overlooking it at the beginning) I've moved the search for the right CMSSW path to `hlt_cmssw_path=$(scram list -c $release_hlt | grep -w $release_hlt | sed 's|.* ||')`
- the tests as they are don't trigger a failure at HLT step unless the `CMSSW_MC_SETUP_TEST_CATCH_HLT` environment variable is set;
- the [tests here](https://github.com/cms-sw/cmssw/pull/44578#issuecomment-20434748842) shows the desired behavior when the variable is set for the 2023 and 2022 MC setup.

AdrianoDee commented:
How can we proceed here? Is it enough just to have the PR open or is better to merge this? In the latter case is there something missing on my side?

makortel commented on Configuration/PyReleaseValidation/test/BuildFile.xml:
```diff
@@ -2,3 +2,33 @@
+-->
+
+<!-- In CMSSW_13_0_18 the auto:phase1_2023_realistic (pre BPix) is 130X_mcRun3_2023_realistic_postBPix_v1 -->
+<test name="test_MC_23_setup" command="test_mc_setup/test_MC_setup.sh 140X_mcRun3_2023_realistic_v3 Run3_2023 2023v12 CMSSW_13_0_18 130X_mcRun3_2023_realistic_v13 Realistic25ns13p6TeVEarly2023Collision">
+</test>
```
These could be
```suggestion
<test name="test_MC_23_setup" command="test_mc_setup/test_MC_setup.sh 140X_mcRun3_2023_realistic_v3 Run3_2023 2023v12 CMSSW_13_0_18 130X_mcRun3_2023_realistic_v13 Realistic25ns13p6TeVEarly2023Collision"/>
```

makortel commented on Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh:
```diff
@@ -0,0 +1,103 @@
+    hlt_cmssw_path=$(scram list -c $release_hlt | grep -w $release_hlt | sed 's|.* ||')
+    echo $hlt_cmssw_path
+fi
+# if [[ ! ("$CMSSW_VERSION" == $release_hlt) ]]; then
+#     hlt_arch=$(echo $(ls -dtrh ${CMS_PATH}${base_arch_no_gcc}*/cms/cmssw/${release_hlt} | tail -1) | cut -d "/" -f 4)
+# else
+#     hlt_arch=$base_arch
+# fi
```
Is this commented out part still useful?

AdrianoDee replied:
Not really, let me remove them.

makortel commented on Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh:
```diff
@@ -0,0 +1,103 @@
+    exit 0
+fi
+
+export SCRAM_ARCH=$base_arch
```
Maybe this line is not necessary anymore? (the ``eval `scram runtime -sh` `` below should set the `SCRAM_ARCH` correctly anyway, right?)

AdrianoDee replied:
Indeed, useless here. Removing.

makortel reviewed (commented):
I'd get this PR merged. It's much easier for others (i.e. us) to test the necessary developments that way. I spotted a few minor things more, otherwise I think this is good to go.

AdrianoDee commented on Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup.sh:
```diff
@@ -0,0 +1,103 @@
+
+# this would make it run only in 14_0_X releases, usefull?
+
+# if [[ ! ("$CMSSW_VERSION" == *"14_0_"*) ]]; then
+#     echo "This is not a 14_0_X release. No need to test this. Exiting."
+#     exit 0
+# fi
```
P.S. this implies that I would backport this to `14_0_X`

makortel replied:
I think as long as there is any chance of a future Run3 UL (or any other future production) where this "HLT step is run in 12_4_X/13_0_X/..., and it needs to read the input from a modern release" behavior is expected, these tests should be kept in `master`. Otherwise it would be sufficient to have the test only in release cycles where this behavior is needed (i.e. as far as I am concerned, if starting from release cycle N this behavior will not be needed, the test can be removed in N).

srimanob replied:
I agree with @makortel that the test should target the master. Backport to 14_0 is special case in this PR.

AdrianoDee replied:
Agreed!

AdrianoDee commented:
code-check

AdrianoDee commented:
I'll leave the next tests to fail, to see if the latest modifications haven't changed the expected behavior. The next commit will be a squash with the failures disabled.

AdrianoDee commented:
please test
- these will fail

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/40015

- This PR adds an extra 20KB to repository

- Found files with invalid states:
  - Configuration/PyReleaseValidation/test/test-MC-setup_hlt.sh:
    - Added: 92cda626a7c55d6aaf6877c74ef38376115cf07c
    - Deleted: fb4419879c023e37639c55d0905a32cd868a79ac
  - Configuration/PyReleaseValidation/test/test_MC_setup.sh:
    - Added: 92cda626a7c55d6aaf6877c74ef38376115cf07c
    - Deleted: fb4419879c023e37639c55d0905a32cd868a79ac
  - Configuration/PyReleaseValidation/test/test-MC-setup_reco.sh:
    - Added: 92cda626a7c55d6aaf6877c74ef38376115cf07c
    - Deleted: fb4419879c023e37639c55d0905a32cd868a79ac
  - Configuration/PyReleaseValidation/test/test-MC-setup_gen_sim.sh:
    - Added: 92cda626a7c55d6aaf6877c74ef38376115cf07c
    - Deleted: fb4419879c023e37639c55d0905a32cd868a79ac

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/40017

- This PR adds an extra 16KB to repository

cmsbuild commented:
Pull request #44578 was updated. @makortel, @cmsbuild, @subirsarkar, @AdrianoDee, @smuzaffar, @miquork, @sunilUIET, @Dr15Jones, @srimanob can you please check and sign again.

AdrianoDee commented:
please test
- will fail (if run)

cmsbuild commented:
Pull request #44578 was updated. @sunilUIET, @subirsarkar, @Dr15Jones, @miquork, @smuzaffar, @makortel, @srimanob, @AdrianoDee, @cmsbuild can you please check and sign again.

AdrianoDee commented:
please test
- these tests will fo smoothly

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10f471/38960/summary.html
**COMMIT**: 9b4d74d14195f04d325508b23f714f1cde2b03b6
**CMSSW**: CMSSW_14_1_X_2024-04-18-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/44578/38960/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 87 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-18-2300+10f471/62442/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 48 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-18-2300+10f471/62442/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3319599
 - DQMHistoTests: [Total failures: 3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-18-2300+10f471/62442/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3319576
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-18-2300+10f471/62442/triggerResults): no differences found

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/40040

- This PR adds an extra 20KB to repository

cmsbuild commented:
Pull request #44578 was updated. @sunilUIET, @AdrianoDee, @makortel, @miquork, @subirsarkar, @cmsbuild, @Dr15Jones, @smuzaffar, @srimanob can you please check and sign again.

AdrianoDee commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10f471/38998/summary.html
**COMMIT**: a7705fb25e2855ebc6ee57ad755a26263cde59e1
**CMSSW**: CMSSW_14_1_X_2024-04-22-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/44578/38998/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 98 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-22-1100+10f471/62465/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 41 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-22-1100+10f471/62465/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3319852
 - DQMHistoTests: [Total failures: 3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-22-1100+10f471/62465/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3319829
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-22-1100+10f471/62465/triggerResults): no differences found

makortel commented:
+core

srimanob commented:
Hi @AdrianoDee 

I am not sure I get the final confirm, is this PR expect to run on master, not only 14_0, right?
From the unit test, I see it runs on master release, should the PR description be updated? 

The current test script is hard-coding GT in 14_0, should it be symbolic one instead for GEN-SIM-DIGI, and RECO. So it is automatically update together with the PR.

srimanob commented on Configuration/PyReleaseValidation/test/BuildFile.xml:
```diff
@@ -2,3 +2,29 @@
+-->
+
+<!-- In CMSSW_13_0_18 the auto:phase1_2023_realistic (pre BPix) is 130X_mcRun3_2023_realistic_postBPix_v1 -->
+<test name="test_MC_23_setup" command="test_mc_setup/test_MC_setup.sh 140X_mcRun3_2023_realistic_v3 Run3_2023 2023v12 CMSSW_13_0_18 130X_mcRun3_2023_realistic_v13 Realistic25ns13p6TeVEarly2023Collision" />
```
I mean this GT, 140X_mcRun3_2023_realistic_v3, should it be phase1_2023_realistic instead? Same for 2022 Iphase1_2022_realistic).

srimanob replied:
The GT for HLT release is more-or-less OK for hard-coding, I think. The same reason as the release.

makortel replied:
Maybe the 14_0_X backport should use the `140X_mcRun3_2023_realistic_v3` GT explicitly?

AdrianoDee replied:
Indeed I think the `14_0_X` would be better to use an hardcoded GT, but also yes for `master` we could use the automatic ones everywhere.

AdrianoDee replied:
Does it sound reasonable?

srimanob replied:
I think using the symbolic GT is a good choice, so we see the error immediately when something was wrong. Note that, '140X_mcRun3_2023_realistic_v3', is also 'phase1_2023_realistic' in 14_X. So carbon-copy backport should be OK also,
https://github.com/cms-sw/cmssw/blob/CMSSW_14_0_X/Configuration/AlCa/python/autoCond.py

AdrianoDee replied:
So you mean keep the HLT running with a different older GT and have it possibly crash (presumably)?

srimanob replied:
Symbolic GT is for GEN-SIM-DIGi and RECO step. Do I misunderstand something?

AdrianoDee replied:
Sure but the scheme would be:

1. GEN-SIM-DIGI with symbolic in `master` (now `141X`);
2. HLT with `130X` or `124X`
3. RECO with symbolic as 1.

Now, once the issues of forward compatibility are solved, wouldn't this setup risk to crash anyway because the GT in (2) is not granted to be compatible with the GT in (1)?

srimanob replied:
What HLT is needed from HLT should not have a big change I assume. At least it keeps the same condition as that year. From Run-2 UL experience, the HLT GT is there for very long.

makortel replied:
> Now, once the issues of forward compatibility are solved, wouldn't this setup risk anyway to crash because the GT in (2) is not granted to be compatible with the GT in (1)?

Wouldn't such a crash be a good thing (by showing the GTs have become incompatible)? (assuming at some point in the future the GTs would have to be made compatible again)

AdrianoDee replied:
Are really GTs for specific years meant (and granted) to be compatible among releases? I have really no clue (@cms-sw/alca-l2?).

AdrianoDee replied:
Thanks Marco. This would settle this (from my point of view).

But, there's a subsequent issue. In `14_0_X` we can't run the GEN-SIM step (at least) with a `130X_mcRun3_2023_realistic_v*` (tested up to the last `v15`)  GT. Since a record needed for  PPS is not there (`CTPPSInterpolatedOpticsRcd`).

```
----- Begin Fatal Exception 24-Apr-2024 18:02:52 CEST-----------------------
An exception of category 'NoRecord' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'RAWSIMoutput_step'
   [2] Prefetching for module PoolOutputModule/'RAWSIMoutput'
   [3] Prefetching for module OscarMTProducer/'g4SimHits'
   [4] Prefetching for module PPSSimTrackProducer/'LHCTransport'
   [5] Calling method for EventSetup module CTPPSInterpolatedOpticalFunctionsESSource/'ctppsInterpolatedOpticalFunctionsESSource'
   [6] While getting dependent Record from Record CTPPSInterpolatedOpticsRcd
Exception Message:
No "LHCInfoPerLSRcd" record found in the EventSetup.

 Please add an ESSource or ESProducer that delivers such a record.
----- End Fatal Exception -------------------------------------------------
```

So in order to run this properly new uniform GTs (`140X` + `130X` GTs for 2023 and 140X` + `124X`   GTs for 2022, I haven't tested for 2022 but I suspect a similar issue) would be needed. Now to get things rolling, would anyway make sense to have this setup merged using the *wrong* GTs at least to have core experts working on the fix for the forward compatibility (that was the original objective here)? With the caveat and a reminder (==issue) that we need new GTs anyway? Or do you prefer to wait for them?

srimanob replied:
Hi @AdrianoDee 
I admit I confuse. The only change I propose is to change the GT used in GEN-SIM-DIGI, and RECO step to symblic one. So, technically, it does not change anything. I don't propose to use 13_0 GT with 14_0, 14_1 release. Do I miss anything?

mmusich commented on Configuration/PyReleaseValidation/test/BuildFile.xml:
```diff
@@ -2,3 +2,29 @@
+-->
+
+<!-- In CMSSW_13_0_18 the auto:phase1_2023_realistic (pre BPix) is 130X_mcRun3_2023_realistic_postBPix_v1 -->
+<test name="test_MC_23_setup" command="test_mc_setup/test_MC_setup.sh 140X_mcRun3_2023_realistic_v3 Run3_2023 2023v12 CMSSW_13_0_18 130X_mcRun3_2023_realistic_v13 Realistic25ns13p6TeVEarly2023Collision" />
```
why the SimBeamSpot cannot be picked up from GT? (option `DBRealistic`)

AdrianoDee replied:
With `140X` it could. But I think that is the mistake here.

We would like to run the production with the original conditions used for the `Summer[22|23]` campaigns (so `124X` and `130X` GTs). And those GTs has no `SimBeamSpotObjectsRcd` included. Now the plan with @cms-sw/alca-l2 would be to have uniform and compatible : 
- `140X` + `130X` GTs for 2023;
- `140X` + `124X`   GTs for 2022.
To have exactly the same conditions + BS picked from GT. 

For the moment to be uniform I would use the `130X` GT for 2023 and `124X` GT for 2022.

makortel commented on Configuration/PyReleaseValidation/test/BuildFile.xml:
```diff
@@ -2,3 +2,29 @@
+<test name="test_MC_23_setup" command="test_mc_setup/test_MC_setup.sh 140X_mcRun3_2023_realistic_v3 Run3_2023 2023v12 CMSSW_13_0_18 130X_mcRun3_2023_realistic_v13 Realistic25ns13p6TeVEarly2023Collision" />
+
+<!-- In CMSSW_13_0_18 the auto:phase1_2022_realistic (pre EE) is 140X_mcRun3_2022_realistic_v3 -->
+<test name="test_MC_22_setup" command="test_mc_setup/test_MC_setup.sh 140X_mcRun3_2022_realistic_v3 Run3 2022v14 CMSSW_12_4_19 124X_mcRun3_2022_realistic_v12 Realistic25ns13p6TeVEarly2022Collision" />
```
12_4_20 was released that includes a ROOT-side fix that is necessary (but not sufficient), so this could be updated to
```suggestion
<test name="test_MC_22_setup" command="test_mc_setup/test_MC_setup.sh 140X_mcRun3_2022_realistic_v3 Run3 2022v14 CMSSW_12_4_20 124X_mcRun3_2022_realistic_v12 Realistic25ns13p6TeVEarly2022Collision" />
```
(not asking strongly, e.g. updating it only if there will be further other updates would be fine for me)

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/40119

- This PR adds an extra 20KB to repository

cmsbuild commented:
Pull request #44578 was updated. @makortel, @subirsarkar, @AdrianoDee, @srimanob, @sunilUIET, @Dr15Jones, @miquork, @smuzaffar, @cmsbuild can you please check and sign again.

AdrianoDee commented:
please test

AdrianoDee commented:
Latest updates (@makortel, @srimanob):
- went for the symbolic GT for all the steps but the HLT. This poses some doubts on how compatible are the 2023/2022 GTs between `master`/`140X` and the target HLT release. But at the moment I have not solution for this. And most probably some action would be needed by AlCa. So, in order to speed things up and to allow core people to check what is needed to have it properly run, I would merge it as is;
- moved to `12_4_20`;
- added a couple of checks for failures given I saw the tests were not failing if any of the intermediate steps hit an error. E.g.: https://github.com/cms-sw/cmssw/blob/ca8c6c1d5654e840e8e1ec55e3827f3922b97b46/Configuration/PyReleaseValidation/test/test_mc_setup/test_MC_setup_gen_sim.sh#L18-L21

AdrianoDee commented:
+pdmv

makortel commented:
Somehow the `test_MC_22_crosscheck` was not run?

AdrianoDee commented:
> Somehow the `test_MC_22_crosscheck` was not run?

Yea sorry, I have just noticed a dummy typo (I had copied the 2023 test twice). Fixed it now. And relaunching the tests.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44578/40125

- This PR adds an extra 20KB to repository

cmsbuild commented:
Pull request #44578 was updated. @subirsarkar, @smuzaffar, @sunilUIET, @cmsbuild, @miquork, @makortel, @srimanob, @AdrianoDee, @Dr15Jones can you please check and sign again.

AdrianoDee commented:
please test

makortel commented:
> * went for the symbolic GT for all the steps but the HLT. This poses some doubts on how compatible are the 2023/2022 GTs between `master`/`140X` and the target HLT release. But at the moment I have not solution for this. And most probably some action would be needed by AlCa.

Given the discussion and outcome from last Friday XC, it seems to me these tests would not really be needed in `master` (as the only use case would be in 14_0_X). So perhaps we keep them in the `master` e.g. until the setup starts working, and then remove them (i.e. leave the setup to be tested only in 14_0_X)? (or am I jumping to that conclusion too quickly?)

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-10f471/39158/summary.html
**COMMIT**: 94e6d612dbc673dadfac80e804606406762bc7a9
**CMSSW**: CMSSW_14_1_X_2024-04-29-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/44578/39158/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



There are some workflows for which there are errors in the baseline:
 24834.78 step 2
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

**Summary**:
 - You potentially [removed 75 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-29-1100+10f471/62600/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 48 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-29-1100+10f471/62600/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3326929
 - DQMHistoTests: [Total failures: 3](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-29-1100+10f471/62600/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3326906
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-04-29-1100+10f471/62600/triggerResults): no differences found

srimanob commented:
> > * went for the symbolic GT for all the steps but the HLT. This poses some doubts on how compatible are the 2023/2022 GTs between `master`/`140X` and the target HLT release. But at the moment I have not solution for this. And most probably some action would be needed by AlCa.
> 
> Given the discussion and outcome from last Friday XC, it seems to me these tests would not really be needed in `master` (as the only use case would be in 14_0_X). So perhaps we keep them in the `master` e.g. until the setup starts working, and then remove them (i.e. leave the setup to be tested only in 14_0_X)? (or am I jumping to that conclusion too quickly?)

I would say we keep the test as we can't predict what will happen :)
Is there a way to keep, but disable the test to run? This is, for example, the test starts to fell in future. Thx.

srimanob commented:
+Upgrade

AdrianoDee commented:
We could simply keep the machinery, the scripts for future but remove the tests for `master` (namely commenting the `BuildFile`) since now we know they work as expected. Then in the back-port to `14_0_X` I'd turn them back on.

makortel commented:
+core

(although I slightly disagree with keeping around commented-out tests, as they could pretty much as easily be resurrected from the history; but that is beyond this PR)

AdrianoDee commented:
+pdmv

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @rappoccio, @sextonkennedy, @antoniovilela (and backports should be raised in the release meeting by the corresponding L2)

AdrianoDee commented:
> (although I slightly disagree with keeping around commented-out tests, as they could pretty much as easily be resurrected from the history; but that is beyond this PR)

A note: for the moment I havent' commented anything. We can still go and remove them once they are no more useful.

rappoccio commented:
+1

* My 0.02CHF: Let's keep the tests around because people may want private tests with 14_1 and we may also need to adjust the strategy if we find any errors with 14_0 reconstruction down the line.
